File: Python/Arrays/Search/BinarySearch.py
def BinarySearch(arr:list[int], searchNumber:int)-> int:
    arr = sorted(arr)
    left : int = 0
    right: int = len(arr) -1
    while( left <= right):
        mid: int = left + (right - left) // 2
        if arr[mid] == searchNumber:
            return mid
        if arr[mid] <= searchNumber:
            left = mid + 1
        else:
            right = mid - 1
    return -1

arr: list[int] = [32,43,92,23,64,2,12,54]
index: int = BinarySearch(arr,54)
if index == -1:
    print(f"The number was not found.")
else:
    print(f"The number was found at index {index}.")

File: TypeScript/Arrays/Search/BinarySearch.ts
const BinarySearch = (arr: number[], searchNumber: number): number => {
  arr = arr.sort((a, b) => a - b);
  let left: number = 0;
  let right: number = arr.length - 1;
  while (left <= right) {
    let mid: number = left + Math.floor((right - left) / 2);
    if (arr[mid] === searchNumber) {
      return mid;
    }
    if (arr[mid] < searchNumber) {
      left = mid + 1;
    } else {
      right = mid - 1;
    }
  }
  return -1;
};

let arr: number[] = [32, 43, 5, 3, 2, 34233, 23];
let index: number = BinarySearch(arr, 23);
if (index === -1) {
  console.log("Number not found.");
} else {
  console.log(`The number is found at the index of ${index}`);
}
